pkg/pxc/backup/restore.go

```diff
@@ -648,3 +649,116 @@ func xbMemoryUse(res corev1.ResourceRequirements) string {
 
 	return useMem
 }
+
+// PrepareJob creates a Kubernetes Job that prepares a restored PXC cluster for operation.
+// This job runs after the data has been restored but before the cluster is started.
+//
+// The job mounts the PVC of the first PXC node and runs preparation scripts
+// to ensure the restored data is ready for the cluster to start properly.
+func PrepareJob(
```

Comment:
If we have time we can add a small unit test for this, mainly to ensure that the specific volumes are always right, and script cmd is set to the right value